playbooks/reboot_cluster.yml

```diff
@@ -1,48 +1,123 @@
 ---
-- hosts: all:!db:!vyatta-*
+- hosts: controller
   serial: 1
   tasks:
   - name: Reboot
     command: shutdown -r now
     async: 0
     poll: 0
-    ignore_errors: true
+    failed_when: false
 
   - name: Wait for SSH to come back
     wait_for: host={{ ansible_default_ipv4.address }}
               port={{ ansible_ssh_port|default(22) }}
               delay=15
     delegate_to: "{{ groups['controller']|first }}"
 
+  - name: cluster health check
+    command: /etc/sensu/plugins/percona-cluster-size.rb -d /root/.my.cnf --expected 3 --criticality critical
+    register: cstat
+    until: cstat | succeeded
+    retries: 5
+    run_once: true
+    delegate_to: "{{ item }}"
+    with_items: "{{ play_hosts }}"
+
 - hosts: db_arbiter
   serial: 1
   tasks:
-  - name: Ensure Garbd is up before rebooting database servers
+  - name: cluster health check
+    command: /etc/sensu/plugins/percona-cluster-size.rb -d /root/.my.cnf --expected 3 --criticality critical
+    register: cstat
+    until: cstat | succeeded
+    retries: 5
+    run_once: true
+    delegate_to: "{{ item }}"
+    with_items: "{{ play_hosts }}"
+
+  - name: reboot
+    command: shutdown -r now
+    async: 0
+    poll: 0
+    failed_when: false
+
+- hosts: compute:!controller:!db_arbiter
+  serial: 1
+  tasks:
+  - name: Reboot
+    command: shutdown -r now
+    async: 0
+    poll: 0
+    failed_when: false
+
+  - name: Wait for SSH to come back
     wait_for: host={{ ansible_default_ipv4.address }}
-              port=4567
+              port={{ ansible_ssh_port|default(22) }}
               delay=15
-    delegate_to: "{{ groups['controller']|first }}"
 
-- hosts: db
+  - name: check nova services
+    command: /etc/sensu/plugins/check-procs.rb -p nova-compute -w 15 -c 30 -W 1 -C 1
+    register: result
+    until: result | succeeded
+    retried: 10
+
+- hosts: swiftnode
+  serial: 1
+  tasks:
+  - name: Reboot
+    command: shutdown -r now
+    async: 0
+    poll: 0
+    failed_when: false
+
+  - name: Wait for SSH to come back
+    wait_for: host={{ ansible_default_ipv4.address }}
+              port={{ ansible_ssh_port|default(22) }}
+              delay=15
+
+  - name: swift health check
+    command: /etc/sensu/plugins/check-swift-dispersion.py
+    register: result
+    until: result | succeeded
+    retries: 10 
+
+- hosts: ceph_osds
   serial: 1
   tasks:
+  - name: Set noout flag
+    command: ceph osd set noout
```

Comment:
I'd check with our ceph folks, but I think you should set this once, then do all the reboots, then set it back. Instead of setting it back/forth each host.